### PR TITLE
AArch64: Add ScratchRegisterDependencyConditions class

### PR DESCRIPTION
**Pull request by Akira1Saitoh**
This commit adds TR_ARM64ScratchRegisterDependencyConditions class
which is a convenience class to temporarily hold register dependencies.

Signed-off-by: Akira Saitoh <saiaki@jp.ibm.com>

### CODE REVIEW

File: compiler/aarch64/codegen/OMRRegisterDependency.hpp [hunk 1]
```diff
@@ -301,4 +301,75 @@ class RegisterDependencyConditions: public OMR::RegisterDependencyConditions
 } // ARM64
 } // OMR
 
+// Convenience class to temporarily hold register dependencies.
+// This class allocates space for the maximum number of GPR and CCR dependencies that an evaluator can use
```
Comment (knn-k) [inline compiler/aarch64/codegen/OMRRegisterDependency.hpp:305]: AArch64 does not have CCRs.

Comment (Akira1Saitoh) [reply compiler/aarch64/codegen/OMRRegisterDependency.hpp]: Fixed

File: compiler/aarch64/codegen/OMRRegisterDependency.cpp [hunk 2]
```diff
@@ -622,3 +622,94 @@ OMR::ARM64::RegisterDependencyConditions::stopUsingDepRegs(TR::CodeGenerator *cg
...
+      if (_gprDeps[i].getRegister() == vr)
+         {
+         // Keep the stronger of the two constraints
+         //
+         TR::RealRegister::RegNum min = std::min(rr, _gprDeps[i].getRealRegister());
+         TR::RealRegister::RegNum max = std::max(rr, _gprDeps[i].getRealRegister());
+         if (min == TR::RealRegister::NoReg)
+            {
+            // Anything is stronger than NoReg
+            _gprDeps[_numGPRDeps].setRegister(vr);
```
Comment (knn-k) [inline compiler/aarch64/codegen/OMRRegisterDependency.cpp:662]: Is the index for `_gprDeps[]` here correct?

Comment (Akira1Saitoh) [reply compiler/aarch64/codegen/OMRRegisterDependency.cpp:662]: This should be `i`. Thanks for pointing it out.

**Comment by knn-k** [comment]
Jenkins build aarch64

**Review by knn-k — APPROVED** [review]

**Pull request merged**